ethcore/src/engines/mod.rs

```diff
@@ -71,6 +71,8 @@ pub trait Engine : Sync + Send {
 	/// Block transformation functions, after the transactions.
 	fn on_close_block(&self, _block: &mut ExecutedBlock) {}
 
+	/// If true, generate_seal has to be implemented.
+	fn seals_internally(&self) -> bool { false }
```

Comment:
I see, but Engine had this sort of logical grouping already. @arkpar suggested to leave it.